dbt/adapters/duckdb/plugins/glue.py

```diff
@@ -132,12 +133,64 @@ def _convert_columns(column_list: Sequence[Column]) -> Sequence["ColumnTypeDef"]
     return column_types
 
 
-def _create_table(client: "GlueClient", database: str, table_def: "TableInputTypeDef") -> None:
+def _create_table(
+    client: "GlueClient", database: str, table_def: "TableInputTypeDef", partition_columns
+) -> None:
     client.create_table(DatabaseName=database, TableInput=table_def)
+    # Create partition if relevant
+    if partition_columns != []:
+        partition_names = [column["Name"] for column in partition_columns]
+        partition_values = [column["Value"] for column in partition_columns]
 
+        partition_location = table_def["StorageDescriptor"]["Location"]
+        for name, value in zip(partition_names, partition_values):
+            partition_location += f"/{name}={value}"
 
-def _update_table(client: "GlueClient", database: str, table_def: "TableInputTypeDef") -> None:
+        partition_input = PartitionInputTypeDef()
+        partition_input["Values"] = partition_values
+        partition_input["StorageDescriptor"] = table_def["StorageDescriptor"]
+        partition_input["StorageDescriptor"]["Location"] = partition_location
+
+        client.create_partition(
+            DatabaseName=database, TableName=table_def["Name"], PartitionInput=partition_input
+        )
+
+
+def _update_table(
+    client: "GlueClient", database: str, table_def: "TableInputTypeDef", partition_columns
+) -> None:
     client.update_table(DatabaseName=database, TableInput=table_def)
+    # Update or create partition if relevant
+    if partition_columns != []:
+        partition_names = [column["Name"] for column in partition_columns]
+        partition_values = [column["Value"] for column in partition_columns]
+
+        partition_location = table_def["StorageDescriptor"]["Location"]
+        for name, value in zip(partition_names, partition_values):
+            partition_location += f"/{name}={value}"
```

Comment:
whatever the logic ends up being for this partition path construction, let's move it to a method so we can unit test it and not repeat it in create/update table

Reply:
Done